server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/rfc/model/RequestObject.scala

```diff
@@ -0,0 +1,43 @@
+/** **************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ * *
+ * http://www.apache.org/licenses/LICENSE-2.0                 *
+ * *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ * ***************************************************************/
+
+package org.apache.james.jmap.rfc.model
+
+object RequestObject {
+
+  import play.api.libs.json._
+
+  case class Capability(value: String) extends AnyVal
+  case class ClientId(value: String) extends AnyVal
+  case class ServerId(value: String) extends AnyVal
+  case class Invocation(value: JsArray) extends AnyVal
```

Comment:
Not sure it takes only an array (https://jmap.io/spec-core.html#the-invocation-data-type)

> 1. A String name of the method to call or of the response.
> 2. A String[*] object containing named arguments for that method or response.
> 3. A String method call id: an arbitrary string from the client to be echoed back with the responses emitted by that method call (a method may return 1 or more responses, as it may make implicit calls to other methods; all responses initiated by this method call get the same method call id in the response).

```scala
case class Invocation(methodName: String, arguments: Array[String], methodCallId: String) 
```

looks more valid to me

Reply:
`Invocation` now had properties but it should contain value except `arguments` are object